underfs/adl/src/main/java/alluxio/underfs/adl/AdlUnderFileSystem.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.adl;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.PropertyKey;
+import alluxio.underfs.UfsFileStatus;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.hdfs.HdfsUnderFileSystem;
+import alluxio.underfs.options.FileLocationOptions;
+
+import com.google.common.base.MoreObjects;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * An {@link UnderFileSystem} uses the Microsoft Azure Blob Storage.
+ */
+@ThreadSafe
+public class AdlUnderFileSystem extends HdfsUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(AdlUnderFileSystem.class);
+
+  /** Constant for the adl URI scheme. */
+  public static final String SCHEME_INSECURE = "adl://";
+
+  /** Constant for the adls URI scheme. */
+  public static final String SCHEME_SECURE = "adls://";
+
+  /**
+   * Prepares the configuration for this Adl as an HDFS configuration.
+   *
+   * @param conf the configuration for this UFS
+   * @return the created configuration
+   */
+  //// @TODO: modify this
+  public static Configuration createConfiguration(UnderFileSystemConfiguration conf) {
+    Configuration adlConf = HdfsUnderFileSystem.createConfiguration(conf);
+    for (Map.Entry<String, String> entry : conf.toMap().entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (PropertyKey.Template.UNDERFS_AZURE_CLIENT_ID.matches(key)) {
+        adlConf.set(key, value);
+      }
+      if (PropertyKey.Template.UNDERFS_AZURE_CLIENT_SECRET.matches(key)) {
+        adlConf.set(key, value);
+      }
+      if (PropertyKey.Template.UNDERFS_AZURE_REFRESH_URL.matches(key)) {
+        adlConf.set(key, value);
+      }
+    }
+    adlConf.set("fs.adl.oauth2.access.token.provider.type", "ClientCredential");
+//    adlConf.set("fs.adl.account.ruizhbigcacheadlsgen1.oauth2.access.token.provider.type", "ClientCredential");
```

Comment:
plz remove commented lines if this is not meant to be used

Reply:
fixed